R/checkers.R

```diff
@@ -23,6 +23,91 @@ check_date <- function(df, max_date, call = rlang::caller_env()) {
 }
 
 
+#' Check that R(t) specified for generating simulated data is of sufficient
+#' length
+#'
+#' @param r_in_weeks a vector indicating the R(t) in weeks
+#' @param ot integer indicating the observed time: length of hospital admissions
+#'  calibration time in days
+#' @param nt integer indicating the nowcast time: length of time between last
+#' hospital admissions date and forecast date in days
+#' @param forecast_horizon integer indicating the duration of the forecast in
+#' days e.g. 28 days
+#' @param call Calling environment to be passed to the type checker
```

Comment:
```suggestion
#' @param call Calling environment to be passed to `cli::cli_abort()` for traceback.
```

Reply:
Is your goal with this pattern is to avoid the error message begin:
```
Error in `check_rt_length()`:
```
and instead have it begin:
```
Error in `<function that called check_rt_length()>()`:
```
?

Reply:
I think so, so that in this case since I call this in the `generate_simulated_data()`  function it would tell you there was an error in that function if you passed in an R(t) of the wrong length that would read 
`"The weekly R(t) specifed isn't long enough to produce",
        "infections for the specified observed/calibration time (`ot`)",
        "nowcast period (`nt`), or forecast horizon (`forecast_horizon`)"`

Reply:
That what I thought. In that case, I am fairly confident the pattern is right but that the documentation should be updated per my suggestion above, as what is written does not apply to this case.